TryGetValue is now being used when getting WeaponInfo instances

SirJson opened the pull request:
A YamlException is thrown if TryGetValue fails. Closes #13692.

SirJson committed:
YamlException is now thrown if WeaponInfo can not be found in Ruleset.Weapons

Removed invalid spacing at the end of the line 36 in ThrowsShrapnel

Prevented NullReferenceException in cases where weapons aren't optional

pchote commented on OpenRA.Mods.Cnc/Traits/MadTank.cs:
```diff
@@ -56,8 +56,19 @@ class MadTankInfo : ITraitInfo, IRulesetLoaded, Requires<ExplodesInfo>, Requires
-			DetonationWeaponInfo = rules.Weapons[DetonationWeapon.ToLowerInvariant()];
+			WeaponInfo thumpDamageWeapon;
+			WeaponInfo detonationWeapon;
+			var thumpDmgWeapToLower = ThumpDamageWeapon.ToLowerInvariant();
```
Style nit: avoid removing arbitrary letters from variable names.  Saving 5 characters on a throwaway local variable isn't worth the hit to readability.  `thumpDamageWeaponToLower` and `detonationWeaponToLower` are fine.  If you really want to use shorter names then `thumpName` and `detonationName` would also be fine considering the context.

pchote replied:
I suggest using `(ThumpDamageWeapon ?? string.Empty).ToLowerInvariant()` here and in the other cases where the weapons aren't optional to avoid a `NullReferenceException` if the mod yaml overrides it to be empty (which translates to a null value).

pchote reviewed (commented):
Looks good overall.  Just a couple of small requests:

pchote commented:
Can you please also squash the commits once you're done?

SirJson commented:
Fixed up everything and squashed my commits. I'm using now the more verbose variant for the local variables for more consistency with the rest of the code.

reaperrr commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e9f411d7a35c99aae1b59678851bc7facba28fbc...e0e37db71451200027c87318fdbda62d97470676)
